commands/class-make-block-command.php

```diff
@@ -0,0 +1,187 @@
+<?php
+/**
+ * Make Command class.
+ *
+ * @package Creode Blocks
+ */
+
+/**
+ * Command in WP_CLI to make a new block.
+ */
+class Make_Block_Command {
+	/**
+	 * Creates a new block class within your theme.
+	 *
+	 * ## OPTIONS
+	 *
+	 * <name>
+	 * : The name of the block to create e.g. "Creode Footer".
+	 *
+	 * [--theme=<active-theme>]
+	 * : The slug of the theme to create the block in.
+	 *
+	 * ## EXAMPLES
+	 *
+	 * wp make-block "Creode Footer" --theme=creode
+	 *
+	 * @when after_wp_load
+	 *
+	 * @param array $args List of arguments.
+	 * @param array $optional_args List of optional arguments.
+	 */
+	public function __invoke( $args, $optional_args ) {
+		$theme_base_path = $this->get_theme_base_path( $optional_args );
```

Comment:
The argument $optional_args isn't really optional since there is no default value but it is user here.